frontend/src/view/Home.jsx

```diff
@@ -1,22 +1,47 @@
 import React from 'react';
 import Button from '../components/Button';
 
-function Home() {
-  return (
-    <div
-      style={{
-        display: 'flex',
-        flexDirection: 'column',
-        alignItems: 'center',
-      }}
-    >
-      <div style={{ maxWidth: '500px' }}>
-        <Button>Test</Button>
-        <Button>Test</Button>
-        <Button>Test</Button>
+class Home extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      teachers: [],
+    };
+  }
+
+  componentDidMount() {
+    const teachers = [
+      {
+        name: 'Jan Kowalski',
+      },
+      {
+        name: 'John Smith',
+      },
+      {
+        name: 'Jan Kowalski',
+      },
+    ];
+
+    this.setState({ teachers });
+  }
+
+  render() {
+    const { teachers } = this.state;
+    return (
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+        }}
+      >
+        <div style={{ maxWidth: '500px' }}>
+          { teachers.map((teacher) => <Button redirect={`/teacher/${teacher.name}`}>{ teacher.name }</Button>)}
```

Comment:
There is a problem with that since `teacher.name` has a space in it. So you need to replace space with `%20`. This should do the work: https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/encodeURI

Reply:
Also you forgot the `key`

Reply:
I don't think we need encodeURI, as the react-router-dom does it automatically
![Zrzut ekranu 2019-12-21 o 14 48 10](https://user-images.githubusercontent.com/24841038/71308793-14c14880-2401-11ea-9137-bfb971490055.png)

Reply:
Are you sure it's caused by react-router-dom? I thought this was done by the browser and if we, for some reason, might change the browser in the future this might stop working. That's why I marked it.
But if it's done by react then it's fine.

Reply:
Yes, I'm pretty sure it's done by react-router

Reply:
Okay, so we’re ready to go.